tools/gen_asm_script.sh

```diff
@@ -9,25 +9,30 @@ rm -f "$1/asm_defines_*"
 
 #	Adaptation of 'gen_asm_defines.awk' / 'gen_asm_script.cmd':
 #
-#	1. Display 'asm_defines.o' as a list
-#	2. Look for the '@ASM_DEFINE' pattern
-#	3. Sort for easy reading
-#	4. Discard "1" ('@ASM_DEFINE') pattern
-#	5. Take "2" (offset*) and "3" (0x*) patterns
-#	6. Exploit LF to space (U+0020) behavior and save all patterns "2" and "3"
-#
+#	1. Display 'asm_defines.o' as a list, looking only for
+#	   patterns '@ASM_DEFINE'
+#	2. Sort for easy reading, the result is interpreted as an array or
+#	   worksheet with ' ' as a self-imposed delimiter in the 'cut' command
+#	3. The cut's '-f2,3' discards column "1" (@ASM_DEFINE) and anything
+#	   beyond "3", take "2" (offset* column), the delimiters and
+#	   "3" (hex* column) to be kept
+#	4. After setting 'GAS_OBJSTR', all the next lines (LF) are degraded
+#	   to ' ' and thus all patterns in columns "2" and "3" become a
+#	   predictable text sequence
 GAS_OBJSTR=$(LC_ALL=C grep -a "@ASM_DEFINE" "$2" | sort | cut -d ' ' -f2,3)
 for GAS_CURSTR in ${GAS_OBJSTR}; do
 
-	#	7. If current 'GAS_CURSTR' has a "hex" glob pattern, print
-	if [ "${GAS_CURSTR:0:2}" = "0x" ]; then
+#	5. If the current value doesn't contain a hex glob (pattern matching),
+#	   it must be a "offset* value"
+	if [ "${GAS_CURSTR:0:2}" != "0x" ]; then
 
-		#	9. Print current values, repeat steps 7-9 until EOL
-		echo "%define ${GAS_OFFSET} (${GAS_CURSTR})" >> "$NASM_FILE"
-		echo "#define ${GAS_OFFSET} (${GAS_CURSTR})" >> "$GAS_FILE"
+#		6. Save current value and repeat step 5 on the next value
+		GAS_OFFSET=${GAS_CURSTR}
 	else
 
-		#	8. Save current "offset"
-		GAS_OFFSET=${GAS_CURSTR}
+#		7. Print current values and repeat steps 5-7 on the next value
+#		   until EOL is reached
+		echo "%define ${GAS_OFFSET} (${GAS_CURSTR})" >> "$NASM_FILE"
+		echo "#define ${GAS_OFFSET} (${GAS_CURSTR})" >> "$GAS_FILE"
```

Comment:
Why use a negate here? What was wrong with the previous condition.

Reply:
Just to give consistency to the comments explaining the workflow...